### ISSUE TITLE
TSQL pagination not recognized

### ISSUE THREAD

**Issue by nils-van-zuijlen** [sqlfluff/sqlfluff#6013]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried using the pagination tools described on https://learn.microsoft.com/en-us/sql/t-sql/queries/select-order-by-clause-transact-sql?view=sql-server-ver16#a-specifying-integer-constants-for-offset-and-fetch-values.

I wrote the following query:
```sql
SELECT
    client.reference,
    client.name
FROM client
GROUP BY client.reference, client.name
ORDER BY client.reference
OFFSET %(offset)s ROWS
FETCH NEXT %(limit)s ROWS ONLY
```

### Expected Behaviour

Running sqlfluff lint produces no warning.

### Observed Behaviour

I got the following parse error.
```
L:   7 | P:   1 |  PRS | Line 7, Position 1: Found unparsable section: 'OFFSET 20                                                         
                       | ROWS\nFETCH NEXT 10 ROWS ONLY'
```

### How to reproduce

The following SQL should pass lints, but does not:
```sql
SELECT
    client.reference,
    client.name
FROM client
GROUP BY client.reference, client.name
ORDER BY client.reference
OFFSET 10 ROWS
FETCH NEXT 10 ROWS ONLY
```

### Dialect

tsql

### Version

sqlfluff, version 3.0.7
Python 3.11.8

### Configuration
```ini
[sqlfluff]
exclude_rules = CP02,ST06,RF04
templater = placeholder
dialect = tsql

[sqlfluff:indentation]
allow_implicit_indents = True
indented_ctes = True

[sqlfluff:templater:placeholder]
param_style = pyformat
offset = 20
limit = 10
```
### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by nils-van-zuijlen** [comment]
Just updated sqlfluff to 3.1.0, this is still present.
